src/components/checkbox/checkbox.tsx

```diff
@@ -0,0 +1,61 @@
+import React = require('react');
+const style = require('./checkbox.css');
+
+export interface CheckBoxProps {
+    text: string;
+    value: boolean;
+    boxIcon: React.SFC<CheckBoxIconProps>;
+    tickIcon: React.SFC<CheckBoxIconProps>;
+    onChange: (value: boolean) => any;
+}
+
+export interface CheckBoxIconProps {
+    value?: boolean;
+}
+
+export const CheckBox: React.SFC<Partial<CheckBoxProps>> = (props) => (
+    <div data-automation-id="CHECKBOX_ROOT"
+         onClick={(event) => executeClickHandler(props.onChange!, !props.value)}
+         style={{display:'inline'}}>
+
+        { props.boxIcon!({value: props.value}) }
+
+        { props.value && props.tickIcon!({value: props.value}) }
+
+        <span data-automation-id="CHECKBOX_LABEL" style={{marginLeft: '5px', verticalAlign:'top'}}>{props.text || ''}</span>
+        {/*<input type="checkbox"/>*/}
```

Comment:
this was actually a placeholder for a native "invisible" component that I totally forgot about! Thanks!